pkg/image/graph/edges.go

```diff
@@ -24,11 +24,25 @@ func AddImageStreamRefEdge(g osgraph.MutableUniqueGraph, node *imagegraph.ImageS
 	g.AddEdge(node, imageStreamNode, ReferencedImageStreamGraphEdgeKind)
 }
 
-// AddAllImageStreamRefEdges calls AddImageStreamRefEdge for every ImageStreamTagNode in the graph
+// AddImageStreamImageRefEdge ensures that a directed edge exists between an ImageStreamImage Node and the IS it references
+func AddImageStreamImageRefEdge(g osgraph.MutableUniqueGraph, node *imagegraph.ImageStreamImageNode) {
+	dockImgRef, _ := imageapi.ParseDockerImageReference(node.Name)
+	imageStream := &imageapi.ImageStream{}
+	imageStream.Namespace = node.Namespace
+	imageStream.Name = dockImgRef.Name
+
+	imageStreamNode := imagegraph.FindOrCreateSyntheticImageStreamNode(g, imageStream)
+	g.AddEdge(node, imageStreamNode, ReferencedImageStreamGraphEdgeKind)
```

Comment:
Why re-use the edge kind?  This eliminates any guarantee of node kinds at the ends and violates its godoc.